callstats-java-sdk/src/main/java/io/callstats/sdk/httpclient/CallStatsHttp2Client.java

```diff
@@ -0,0 +1,155 @@
+package io.callstats.sdk.httpclient;
+
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Collections;
+import java.util.List;
+import java.util.Properties;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import io.callstats.sdk.internal.CallStatsConst;
+import io.callstats.sdk.internal.NameValuePair;
+import io.callstats.sdk.internal.listeners.CallStatsHttp2ResponseListener;
+import okhttp3.Call;
+import okhttp3.Callback;
+import okhttp3.ConnectionPool;
+import okhttp3.ConnectionSpec;
+import okhttp3.FormBody;
+import okhttp3.FormBody.Builder;
+import okhttp3.MediaType;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.RequestBody;
+import okhttp3.Response;
+import okhttp3.TlsVersion;
+
+public class CallStatsHttp2Client {
+
+	/** The base url. */
+	private String EVENTS_BASE_URL;
+	private String STATS_BASE_URL;
+	private String AUTH_BASE_URL;
+
+	/** The connection time out. */
+	private int connectionTimeOut = CallStatsConst.CONNECTION_TIMEOUT;
+
+	/** The Constant logger. */
+	private static final Logger logger = LogManager.getLogger("CallStatsAsyncHttpClient");
+
+	private OkHttpClient okHttpClientclient;
+	private boolean isDisrupted;
+
+	public boolean isDisrupted() {
+		return isDisrupted;
+	}
+
+	public void setDisrupted(boolean isDisrupted) {
+		this.isDisrupted = isDisrupted;
+	}
+
+	public CallStatsHttp2Client() {
+		ConnectionSpec spec = new ConnectionSpec.Builder(ConnectionSpec.MODERN_TLS).tlsVersions(TlsVersion.TLS_1_2).allEnabledCipherSuites()
```

Comment:
I'm not absolutely sure if it's safe to make this mandatory as it will at least break backwards-compatibility. In most cases it will work, but if person is using SSL inspecting proxy, it might be possible that it doesn't serve TLS 1.2

Reply:
If no TLS 1.2 go collector will not accept my reqests.